### PR TITLE
fix: dropdown title does not follow search name

### PR DESCRIPTION
**Pull request by dominictb**
### Fixed Issues

$ https://github.com/Expensify/App/issues/49208
PROPOSAL: https://github.com/Expensify/App/issues/49208#issuecomment-2351566936

#### 1. Test case 1: Small screen device:

Precondition: User has saved a few searches.
1. Go to staging.new.expensify.com
2. Go to Search.
3. Tap on the dropdown.
4. Open any saved search. Verify the dropdown title should follow the name of the search.
5. Tap on the dropdown. Verify when viewing a saved search, there should not be "type:expense" tab in the dropdown list.

#### 2. Test case 2: Small screen device:
Precondition: User has saved a few searches.
1. Go to staging.new.expensify.com
2. Go to Search.
3. Tap on the dropdown.
4. Click three dots on any option > Rename > Enter a new name > Save.
5. Verify the dropdown title should follow the new name.

#### 3. Test case 3: Large screen device:
Precondition: User has saved a few searches.
1. Go to staging.new.expensify.com
2. Go to Search.
3. Open any saved search. Verify it works properly as it is because these changes only apply to small screen.

- [x] Verify that no errors appear in the JS console

### QA Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/e9766403-35bc-47f1-94e7-b1d47adc4ff3

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/30598058-6135-456b-991a-3314d3f114eb

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0f07f54c-6690-469f-b168-54c895ac2ab8

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/87761e43-ce5a-456f-8e84-cd39f10f87e6

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/7247f321-88e9-4d99-8d48-e4f230f3733d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/4bc39469-a47b-4fc4-b612-8aed5f2e0ff7

</details>

### CODE REVIEW

**Comment by brunovjk** [comment]
When renaming, the field is always empty, is this a bug already reported? Thanks.


https://github.com/user-attachments/assets/9a24a0f3-1eb4-41a3-9207-122391669dc6

@dominictb @lakchote

**Comment by lakchote** [comment]
> When renaming, the field is always empty, is this a bug already reported? Thanks.
> 
> @dominictb @lakchote

This is not a bug. This is what was shown in the design template, however there's discussion around it to include it.

There's an [issue](https://github.com/Expensify/App/issues/49207) for that.

**Comment by brunovjk** [comment]
> > When renaming, the field is always empty, is this a bug already reported? Thanks.
> > @dominictb @lakchote
> 
> This is not a bug. This is what was shown in the design template, however there's discussion around it to include it.
> 
> There's an [issue](https://github.com/Expensify/App/issues/49207) for that.

Thanks for clarifying, I will continue with the checklist.

**Comment by brunovjk** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



https://github.com/user-attachments/assets/eaf2adda-0bf9-4fd1-93d7-25998ed595e3


</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/user-attachments/assets/a675976d-d143-49a9-b084-5a14a9dbe933


</details>

<details>
<summary>iOS: Native</summary>



https://github.com/user-attachments/assets/e7ef8494-37b3-44c9-bae4-f53775210e41


</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/fd496254-7695-4e77-a31b-5c18792c50e6


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/ff1de6e1-82c6-4482-9725-802254c74097


</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/user-attachments/assets/0f198e56-0e94-4a1a-9e8a-27ae8552432c


</details>

**Review by brunovjk — APPROVED** [review]
Looking good to me :D

**Comment by lakchote** [comment]
> Tap on the dropdown. Verify when viewing a saved search, there should not be "type:expense" tab in the dropdown list.

This condition is falsy in your iOS test @brunovjk

**Comment by brunovjk** [comment]
> > Tap on the dropdown. Verify when viewing a saved search, there should not be "type:expense" tab in the dropdown list.
> 
> This condition is falsy in your iOS test @brunovjk

Hi @lakchote, thanks for your feedback. You are right. I first assumed the focus of this PR was maintaining consistency of saved search names. But indeed, I think we can update `getSearchHeaderTitle` to avoid always returning `type:...`:

https://github.com/Expensify/App/blob/33debc8c1c1a64b386f7b236593a9896c33965bb/src/libs/SearchUtils.ts#L748-L758

cc: @dominictb @luacmartins

**Comment by brunovjk** [comment]
I uploaded the wrong video from Android Native, I will post the correct one in a moment.

**Review by luacmartins — COMMENTED** [review]
Code is looking good. I think we should also just remove the custom menu item, if we're viewing a saved search. cc @Expensify/design 

Screenshot 2024-09-20 at 8 27 10 AM

**Comment by shawnborton** [comment]
Agree with that. Also, shouldn't the saved searches have a bookmark icon?

**Comment by luacmartins** [comment]
Yes, that was reported in another issue

**Comment by lakchote** [comment]
@dominictb can you modify your PR to address the recent comments?

**Comment by luacmartins** [comment]
@dominictb bump on the comments above

**Comment by luacmartins** [comment]
@dominictb we should remove the canned query entry from the dropdown
Screenshot 2024-09-23 at 1 23 02 PM

**Comment by luacmartins** [comment]
No, we should only remove it when viewing a saved search

**Comment by lakchote** [comment]
cc @dominictb 

friendly bump for @brunovjk's comment [here](https://github.com/Expensify/App/pull/49446#issuecomment-2368513620)

**Comment by lakchote** [comment]
> > This condition is falsy in your iOS test @brunovjk
> 
> @brunovjk Sorry, but I noticed that in all your iOS videos in the [checklist](https://github.com/Expensify/App/pull/49446#issuecomment-2361187743), it works fine.
> 
> > I think we can update getSearchHeaderTitle to avoid always returning type:... status
> 
> @brunovjk Can you share your solution? IMO, I think with the search that does not have a custom name, using "type:...status" format is fine.

cc @brunovjk in case you've missed this

**Comment by brunovjk** [comment]
Thanks :D I didn't miss it, I'm finishing another PR and I'll be back here later today.

**Comment by brunovjk** [comment]
> ### Details
> ### Fixed Issues
> $ #49208 PROPOSAL: [#49208 (comment)](https://github.com/Expensify/App/issues/49208#issuecomment-2351566936)
> 
> ### Tests
> #### 1. Test case 1: Small screen device:
> Precondition: User has saved a few searches.
> 
> 1. Go to staging.new.expensify.com
> 2. Go to Search.
> 3. Tap on the dropdown.
> 4. Open any saved search. Verify the dropdown title should follow the name of the search.
> 5. Tap on the dropdown. Verify when viewing a saved search, there should not be "type:expense" tab in the dropdown

I believe what @lakchote meant [here](https://github.com/Expensify/App/pull/49446#issuecomment-2361261588) @dominictb is that we are failing in the step above this PR itself.

What I tested here was _"Verify the dropdown title should follow the name of the search"_ and _"remove the canned query entry from the dropdown"_. However, there is still one missing point in the Expected Result of the [issue](https://github.com/Expensify/App/issues/49208):

> In Step 5, when viewing a saved search, there should not be "type:expense" tab in the dropdown list.

I believe the expected name when we create a "search" is something like the videos of this [PR](https://github.com/Expensify/App/pull/48566). @lakchote  and @luacmartins  what do you think? Does that make sense? Thank you very much.

**Comment by luacmartins** [comment]
>I believe the expected name when we create a "search" is something like the videos of this https://github.com/Expensify/App/pull/48566, without type:... status:...by default. @lakchote and @luacmartins what do you think? Does that make sense? Thank you very much.

I think these two should match and IMO it seems valuable to leave the type and status there, especially for saved searches since that could be saved for any type/status

Screenshot 2024-09-26 at 9 38 14 AM

**Comment by lakchote** [comment]
> > I believe what @lakchote meant [#49446 (comment)](https://github.com/Expensify/App/pull/49446#issuecomment-2361261588) @dominictb is that we are failing in the step above this PR itself.
> 
> @lakchote Can you give me more detail, on what is failing there since you mentioned _"This condition is falsy in your iOS test @brunovjk"_

There seems to be confusion on what was expected.

The problem I was referring (which @luacmartins talked about too [here](https://github.com/Expensify/App/pull/49446#issuecomment-2367217165)) was solved already by @dominictb.

When viewing a saved search, we should not see in the dropdown the original query of the saved search, we should hide it.

We should not try to hide specifically`type:... status :..` (for the reasons mentioned [here](https://github.com/Expensify/App/pull/49446#issuecomment-2375509391)), as such @dominictb's PR seems to be looking good to me.

cc @brunovjk

**Comment by brunovjk** [comment]
I see now, yes I did get confusion, sorry, thanks for clarifying @lakchote, I will retest and update Screenshots/Videos now.

**Comment by brunovjk** [comment]
LGTM, all yours @luacmartins.

**Review by luacmartins — APPROVED** [review]

File: src/pages/Search/SearchPageBottomTab.tsx [hunk 1]
```diff
@@ -28,6 +28,7 @@ function SearchPageBottomTab() {
     const [selectionMode] = useOnyx(ONYXKEYS.MOBILE_SELECTION_MODE);
     const searchParams = activeCentralPaneRoute?.params as AuthScreensParamList[typeof SCREENS.SEARCH.CENTRAL_PANE];
     const parsedQuery = SearchUtils.buildSearchQueryJSON(searchParams?.q);
+    const searchName = searchParams?.name;
```
Comment (suneox) [inline src/pages/Search/SearchPageBottomTab.tsx:31]: This change misses covering the case where the search name equals the query name which introduce [this issue](https://github.com/Expensify/App/issues/50223) so we should get searchName based on search query

**Pull request merged**